core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -450,9 +452,43 @@ public String name()
     }
 
     @Override
-    protected ExprEval eval(double param)
+    public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
     {
-      return ExprEval.of(Math.round(param));
+      if (args.size() == 1) {
+        ExprEval value1 = args.get(0).eval(bindings);
+        if (value1.type() != ExprType.LONG && value1.type() != ExprType.DOUBLE) {
+          throw new IAE("first argument should be integer or double type but got %s type", value1.type());
+        }
+        return eval(value1);
+      } else if (args.size() == 2) {
+        ExprEval value1 = args.get(0).eval(bindings);
+        ExprEval value2 = args.get(1).eval(bindings);
+        if (value1.type() != ExprType.LONG && value1.type() != ExprType.DOUBLE) {
+          throw new IAE("first argument should be integer or double type but got %s type", value1.type());
```

Comment:
It would be better to add the function name to this error message. For example, `The first argument to the function[%s] should be integer or double type but get the %s type`.